src/common/transformations/tests/op_conversions/fake_convert_decomposition_test.cpp

```diff
@@ -0,0 +1,153 @@
+                                                  Shape,            // scale shape
+                                                  Shape,            // shift shape
+                                                  element::Type_t,  // input precision
+                                                  element::Type_t,  // destination precision
+                                                  bool>;            // default shift
+
+class FakeConvertDecompositionTest : public ov::test::TestsCommon,
+                                     public ::testing::WithParamInterface<FakeConvertDecompositionParams> {
+public:
+    static std::string getTestCaseName(::testing::TestParamInfo<FakeConvertDecompositionParams> obj) {
+        FakeConvertDecompositionParams params = obj.param;
+
+        Shape data_shape, scale_shape, shift_shape;
+        element::Type_t data_prec, dst_prec;
+        bool default_shift;
+        std::tie(data_shape, scale_shape, shift_shape, data_prec, dst_prec, default_shift) = params;
+
+        std::ostringstream result;
+        result << "dataShape=" << ov::test::utils::vec2str(data_shape) << "_";
+        result << "scaleShape=" << ov::test::utils::vec2str(scale_shape) << "_";
+        result << "shiftShape=" << ov::test::utils::vec2str(shift_shape) << "_";
+        result << "dataPrecision=" << element::Type(data_prec) << "_";
+        result << "destinationPrecision=" << element::Type(dst_prec) << "_";
+        if (default_shift)
+            result << "destinationPrecision=true";
+        else
+            result << "defaultShift=false";
+        return result.str();
+    }
+
+protected:
+    void SetUp() override {
+        FakeConvertDecompositionParams params = this->GetParam();
+
+        Shape data_shape, scale_shape, shift_shape;
+        element::Type_t data_prec, dst_prec;
+        bool default_shift;
+        std::tie(data_shape, scale_shape, shift_shape, data_prec, dst_prec, default_shift) = params;
+
+        std::shared_ptr<ov::Model> f(nullptr);
+        {
+            const auto data = std::make_shared<opset1::Parameter>(data_prec, PartialShape(data_shape));
+            const auto scale = std::make_shared<opset1::Constant>(data_prec, scale_shape);
+            const auto shift = std::make_shared<opset1::Constant>(data_prec, shift_shape);
+
+            const auto fake_convert = default_shift
+                                          ? std::make_shared<opset13::FakeConvert>(data, scale, dst_prec)
+                                          : std::make_shared<opset13::FakeConvert>(data, scale, shift, dst_prec);
+            f = std::make_shared<ov::Model>(NodeVector{fake_convert}, ParameterVector{data});
+
+            pass::Manager manager;
+            manager.register_pass<ov::pass::InitNodeInfo>();
+            manager.register_pass<ov::pass::FakeConvertDecomposition>();
+            manager.run_passes(f);
+
+            OV_ASSERT_NO_THROW(check_rt_info(f));
```

Comment:
Can test be added in test body instead test's setup?

Reply:
Applied. Thanks Pawel!